src/ServicePulse.Host/vue/src/components/monitoring/EndpointList.vue

```diff
@@ -140,22 +140,22 @@ function formatGraphDecimal(input, deci) {
               <div class="col-xs-2 col-xl-7 endpoint-name name-overview">
                 <div class="row box-header">
                   <div class="col-lg-max-3 no-side-padding lead righ-side-ellipsis endpoint-details-link">
-                    <a @click="navigateToEndpointDetails($event, endpoint.name)" class="cursorpointer" :title="`${endpoint.name}`">
+                    <a @click="navigateToEndpointDetails($event, endpoint.name)" class="cursorpointer" v-tooltip :title="endpoint.name">
                       {{ endpoint.name }}
                     </a>
                   </div>
-                  <span class="endpoint-count ng-binding ng-scope" v-if="endpoint.connectedCount || endpoint.disconnectedCount" :title="`Endpoint instance(s): ${endpoint.connectedCount || 0} `">({{ endpoint.connectedCount || 0 }})</span>
+                  <span class="endpoint-count ng-binding ng-scope" v-if="endpoint.connectedCount || endpoint.disconnectedCount" v-tooltip :title="`Endpoint instance(s):`+ endpoint.connectedCount || 0 ">({{ endpoint.connectedCount || 0 }})</span>
                   <div class="col-xs-5 no-side-padding endpoint-status">
-                    <span class="warning" ng-if="endpoint.metrics.criticalTime.displayValue < 0">
-                      <i class="fa pa-warning" :title="`Warning: endpoint currently has negative critical time, possibly because of a clock drift.`"></i>
+                    <span class="warning" v-if="endpoint.metrics!=null && formatGraphDuration(endpoint.metrics.criticalTime).value < 0">
```

Comment:
can these conditionals get moved to a computed function with a meaningful name?

Reply:
Yes definitely, it can be optimized to a computed field. There are other such that can be optimized. So, will address it another PR